### DISCUSSION TITLE
snake_case vs CamelCase

### DISCUSSION THREAD

**Discussion by jordonezlucena** [camaraproject/WorkingGroups#169] [Polls]
The [API design guidelines documentation](https://github.com/camaraproject/WorkingGroups/blob/main/Commonalities/documentation/API-design-guidelines.md) notes the use of snake_case for notation. However, some companies prefer to change to camelCase notation -- see discussion in issue #157.

The proposed poll is to decide the preferred notation style: snake_case or camelCase. CAMARA APIs will need to adhere to the winning option. 

**Instructions**: one single vote per participant company.
**Deadline**: Thursday 16th March

**Comment by eric-murray** [comment]
It would appear that anyone with a github account can vote in these polls, and the results are anonymous.

In the interests of transparency, and to attempt to maintain the one company one vote policy, I'd suggest that anyone who votes puts their vote in the comments.

Vodafone are voting for camelCase.

**Reply by GarethWGSMA** [reply]
Hi I'm not sure if the members of GSMA get a vote, but if we do I vote **C**amelCase (Upper Camel case). I find a capital at the start easier to read than lowercase. It "jumps out" at me in url's etc ... but I won't make an issue if the votes go against me.

**Reply by eric-murray** [reply]
Hi @GarethWGSMA. The GSMA are participating members of CAMARA, so you get a vote. But you won't be able to vote for CamelCase!

**Reply by sharrop** [reply]
Hi @GarethWGSMA : We should clarify: I think there is no problem/debate over resource names being UpperCamelCase. The [CAMARA guidelines already state this](https://github.com/camaraproject/WorkingGroups/blob/main/Commonalities/documentation/API-design-guidelines.md#41-url-definition):
> _OperationIds are defined in lowerCamelCase: For example: helloWorld_
> _Objects are defined in CamelCase inside properties field. For example: Greetings, ExampleObject._

This vote is specifically about the attribute/field/property names _within_ the objects/resources.

**Comment by shilpa-padgaonkar** [comment]
DT votes for camelCase

**Comment by GarethWGSMA** [comment]
Fair enough. Then I will vote camelCase

Regards,

Gareth Williams
Technical Architect
***@***.***<https://www.linkedin.com/in/rgarethwilliams>

***@***.***

**Comment by sfnuser** [comment]
Spry Fox Networks votes for camelCase

**Comment by ajamous** [comment]
TelecomsXChange votes camelCase

**Comment by tanjadegroot** [comment]
Nokia votes for camelCase for consistency with many of the standards APIs that CAMARA is abstracting on top of.

**Comment by bigludo7** [comment]
Orange votes for CamelCase.

**Comment by jordonezlucena** [comment]
Telefónica votes snake_case, among other reasons, to avoid having a mix of cases, as OAuth/OIDC is snake_case
